SEIR/NPI/Reduce.py

```diff
@@ -100,10 +100,11 @@ def __createFromDf(self, loaded_df):
         self.param_name = self.parameters["parameter"].unique()
 
     def getReduction(self, param):
+        "Return the reduction for this param, None if no reduction defined"
+
         if param == self.param_name:
             return self.npi
```

Comment:
Can we make self.npi lazy instead of the stacked?  Should reduce memory the same amount I would expect.  Am I missing something?

Reply:
i'm not sure i see what making construction happen at access (lazy) would do?

Reply:
sorry, i meant on demand (i.e., the Reduce interventions don't allocate a self.npi, and only allocate one when it's used)

Reply:
that's a good point. i first made construction of the npi on access (vs on construction). this had small savings (we use the matrix right after construction), the real perf benefits are around: reducing the number of `O(|geoids| * |dates|)`-sized matrices and operations on them (e.g., we had 1 for every potential parameter that could be reduced), pre-fetching config, limiting the amount of debug npis we keep around, etc.